assets/source/setup-guide/app/steps/ClaimWebsite.test.js

```diff
@@ -48,4 +57,25 @@ describe( 'Claim Website Record Events', () => {
 			expect.any( Object )
 		);
 	} );
+
+	it( '`pfw_domain_verify_success` is fired when a site is successfully verified', async () => {
+		apiFetch.mockImplementation( () => {
+			return { account_data: { id: 'foo' } };
+		} );
+
+		const { getByText } = render(
+			<ClaimWebsite goToNextStep={ () => {} } view="wizard" />
+		);
+
+		fireEvent.click( getByText( 'Start verification' ) );
+
+		// Wait for async click handler and apiFetch resolution.
+		const delay = sleep( 1 );
+		jest.runAllTimers();
+		await delay;
+
+		expect( recordEvent ).toHaveBeenCalledWith(
+			'pfw_domain_verify_success'
+		);
```

Comment:
💅 

WDYT about...

```
await waitFor( () =>
			expect( recordEvent ).toHaveBeenCalledWith(
				'pfw_domain_verify_success'
			)
		);
```

more info [here](https://testing-library.com/docs/dom-testing-library/api-async/#waitfor)

Reply:
I like it :)
`setTimout(1)` was testing that it's fired immediately after, but `waitFor` should be less fragile` is more readable, and shorter.

Done in https://github.com/woocommerce/pinterest-for-woocommerce/pull/314/commits/6fac7240f530b4c2546ad69566e63eb92acaf04b

We could think of something more like
```js
await waitForNetworkIdle(); // await waifForApiFetchMock()
expect( //...
```
to make sure we fire it synchronously after  API fetch response. But let's not overdo it.